### DISCUSSION TITLE
WebUSB Support

### DISCUSSION THREAD

**Discussion by mepunit** [coder/code-server#2892] [Ideas] [answered]
Hello Code-Server Team,

I need WebUSB Support inside the Code-Server environment. I want to access some USB devices and want to develop a VSCode Plugin to communicate with the device.

But, strangely, I am not able to access navigator.serial or navigator.usb within the DOM which is accessible otherwise.

Please help me on this.

Thank you,
Punit

**Comment by bpmct** [comment, marked as answer]
@code-asher might be able to help with this. Any idea on the best way to access the DOM from within code-server?

**Reply by mepunit** [reply]
@bpmct Thank you Ben. But, I just now checked I was facing this issue because I was using an Unsecured Docker Instance. When I used SSL, the problem vanished.

**Reply by bpmct** [reply]
Glad you got it sorted! Feel free to add updates about the WebUSB plugin. Would love to document this!

**Reply by nooobcoder** [reply]
It seems amazing, 80% of problems in development are just because of SSL certificates, lol 😂
code-server's discussions are flooded with problems that were due to HTTP.
